picoquic/tls_api.c

```diff
@@ -956,6 +964,69 @@ Decrypt returns size_t_max (-1) if decryption fails, number of bytes in output o
 
 */
 
+int picoquic_hkdf_expand_label(ptls_hash_algorithm_t *algo, void *output, size_t outlen, ptls_iovec_t secret, 
+    const char *label, const char *base_label)
+{
+    ptls_buffer_t hkdf_label;
+    uint8_t hkdf_label_buf[512];
+    int ret;
+
+    ptls_buffer_init(&hkdf_label, hkdf_label_buf, sizeof(hkdf_label_buf));
+
+    ptls_buffer_push16(&hkdf_label, (uint16_t)outlen);
+    ptls_buffer_push_block(&hkdf_label, 1, {
+        if (base_label == NULL)
+        base_label = "tls13 ";
```

Comment:
If the `base_label` is not given, it should be "QUIC". However, as the function now exists in our own code, we do not need to give the `base_label` as argument as it is fixed in quic :) So, I would remove this parameter.

Reply:
I thought about that, but I would rather wait until the current discussion of https://github.com/quicwg/base-drafts/issues/1255. If the format reverts to being TLS compatible, we will have
to revert this change. If it doesn't, there are other things that can be cleaned up, e.g. the construction of label for plaintext AEAD, and the generation of PN encryption keys. I added issue #149 to track that.